elementary/files
# Show path in recent

Because items in recent:// 'folder' are allowed to have the same name, this PR disambiguates them by using the path (up to a maximum length) as the display name.

Also SearchResults is modified so that the original folder of files is displayed, to disambiguate them.  Clicking on a search result now navigates to the actual file, not the item in recent://

TO TEST:

* If you already have files in your Recent folder then you should see the effect straight away in ListView and SearchBar.  
* Create two files with the same (best to be unusual) name in different folders which also have the same name then search for that file name.  You will see that those search results are distinguishable now.
* If you have History 'off' in Switchboard Privacy plug, then instead of Recent view you will get a Welcome view with an appropriate message and an opportunity to open Switchboard Privacy plug.
* Create two files as above then edit them with an app that records history (note Code does not seem to do this for some reason). Or play appropriately named videos or music files (depending on your settings - these will also appear in Recent).

You can run the PR from the build directory without installing.  Run master Files from Plank for comparison (one at a time).

Needs UX team opinion on new appearance of Recent VIew.
